### PR TITLE
pythonPackages.qds_sdk: init at 1.12.0

### PR DESCRIPTION
**Pull request by shahrukh330**
###### Motivation for this change
Add Qubole Data Service Python SDK to nixpkgs. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @kalbasit

### CODE REVIEW

File: pkgs/development/python-modules/qds_sdk/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  version = "1.12.0";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "qds_sdk";
+    sha256 = "113qrfz165sqcbndgdpvr4q7p08bdfw53sh1wrkvip8g2cngh6p3";
+  };
+
+  # No tests available
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/qds_sdk/default.nix]: since they don't distribute tests with their sdist pypi package, we could also pull from github: https://github.com/qubole/qds-sdk-py.

Tests are pretty import to ensure the the package isn't going to be broken when upstream dependencies get bumped :)

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -5807,6 +5807,12 @@
     githubId = 1151264;
     name = "Sebastian Graf";
   };
+  shahrukh330 = {
+    email = "shahrukh330@gmail.com";
+    github = "shahrukh330";
+    githubId = 1588288;
+    name = "Shahrukh Khan";
+  };
```
Comment (kalbasit) [inline maintainers/maintainer-list.nix]: @shahrukh330 please put this in a separate commit. Your PR should have one commit for the new maintainer, and another for the qds_sdk init (current commit).

File: pkgs/development/python-modules/qds_sdk/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  checkInputs = [ pytest ];
+  propagatedBuildInputs = [ requests boto six urllib3 inflection];
+
+  meta = with lib; {
+    description = "A Python module that provides the tools you need to authenticate with, and use the Qubole Data Service API";
+    homepage = https://github.com/qubole/qds-sdk-py;
```
Comment (kalbasit) [inline pkgs/development/python-modules/qds_sdk/default.nix]:
```suggestion
    homepage = "https://github.com/qubole/qds-sdk-py";
```

Ref: https://github.com/NixOS/rfcs/pull/45

File: pkgs/development/python-modules/qds_sdk/default.nix (hunk 1 above)
Comment (shahrukh330) [reply pkgs/development/python-modules/qds_sdk/default.nix]: Thank you for the review, I have updated the code to use `fetchgit` and run the tests.

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (shahrukh330) [reply maintainers/maintainer-list.nix]: Reverted the previous commit and added two separate commits.

File: pkgs/development/python-modules/qds_sdk/default.nix (hunk 3 above)
Comment (shahrukh330) [reply pkgs/development/python-modules/qds_sdk/default.nix]: Done.

File: pkgs/development/python-modules/qds_sdk/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ requests,
+ six,
+ urllib3 }:
+
+buildPythonPackage rec {
+  pname = "qds_sdk";
+  version = "1.12.0";
+
+  # pypi does not contain tests, using github sources instead
+  src = fetchgit {
```
Comment (kalbasit) [inline pkgs/development/python-modules/qds_sdk/default.nix]: Please use `fetchFromGitHub` instead of `fetchgit`

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (kalbasit) [reply maintainers/maintainer-list.nix]: Please force-push to remove the previous commit and the revert. The PR should only contain two commits:
- maintainers: add shahrukh330
- pythonPacakges.qds_sdk: init at 1.12.0

Thank you!

File: pkgs/development/python-modules/qds_sdk/default.nix (hunk 1 above)
Comment (kalbasit) [reply pkgs/development/python-modules/qds_sdk/default.nix]: Resolving this as dupe of https://github.com/NixOS/nixpkgs/pull/70055#discussion_r330065091.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2Packages.qds_sdk python3Packages.qds_sdk

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2Packages.qds_sdk python3Packages.qds_sdk

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2Packages.qds_sdk python3Packages.qds_sdk

File: pkgs/development/python-modules/qds_sdk/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    requests
+    six 
+    urllib3 
+  ];
+
+  checkInputs = [ pytest mock ];
+  checkPhase = ''
+    py.test --disable-pytest-warnings tests
+  '';
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/qds_sdk/default.nix]: minor point, and doesn't really matter, but `buildPythonPackage` will default doCheck to true. This line shouldn't be needed.
```suggestion
```

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2Packages.qds_sdk python3Packages.qds_sdk

**Review by kalbasit — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
